src/custom/pages/NewSwap/index.tsx

```diff
@@ -15,6 +15,20 @@ interface NewSwapPageProps {
   allowedSlippage: Percent
 }
 
+function swapPagePropsChecker(prev: NewSwapPageProps, next: NewSwapPageProps): boolean {
+  const isInputCurrencyEqual =
+    prev.currencies.INPUT && next.currencies.INPUT
+      ? prev.currencies.INPUT.equals(next.currencies.INPUT)
+      : prev.currencies.INPUT === next.currencies.INPUT
```

Comment:
If any is unset, can't you return `false` straight away?

Or do you need to compare if both are undefined or `''`?

Reply:
I can use `equals()` method only if both are set, otherwise I just compare them using ===